docs/develop/node/compile-and-run-a-node.md

```diff
@@ -0,0 +1,265 @@
+- `data/` folder in which NEAR node will write the state.
+
+Initially the required working directory needs to be generated. The command to generate it looks like:
+
+```bash
+$ ./target/release/neard --home ~/.near init --chain-id testnet --download
+```
+
+> You can skip `--home` argument if you are fine with default working directory in `~/.near` or pass different path
+
+This command will create the required directory structure, will generate `config.json` and `node_key.json`, will download `genesis.json` for testnet network.
+
+> **Heads up**
+> The genesis file for the testnet is big (6GB +) so this command will be running for a while, no progress will be shown.
+
+### 4. Replacing the `config.json`
+
+Generated `config.json` is missing `boot_nodes` parameter (it is empty) so we will need to replace it by full one. The file can be downloaded:
+
+- Download `config.json` https://s3-us-west-1.amazonaws.com/build.nearprotocol.com/nearcore-deploy/testnet/config.json and replace it in your working dir (`~/.near/config.json`)
+
+Or run the commands:
+
+```bash
+$ rm ~/.near/config.json
+$ wget ~/.near/config.json https://s3-us-west-1.amazonaws.com/build.nearprotocol.com/nearcore-deploy/testnet/config.json
+```
+
+### 5. Get data backup
+
+The node is ready to be started. However the first thing to do is to sync up with the netowk. It means your node needs to download all the headers and blocks like other nodes in the network have. This process might be speed up drastically by using backups.
+
+Download and unpack the tar to `~/.near` https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/testnet/rpc/data.tar
+
+Or run the commands
+
+```bash
+$ wget ~/.near/data.tar https://near-protocol-public.s3.ca-central-1.amazonaws.com/backups/testnet/rpc/data.tar
+$ tar -xf ~/.near/data.tar
+$ rm ~/.near/data.tar
+```
+
+### 6. Run a node
+
+```bash
+$ ./target/release/neard --home ~/.near run
+```
+
+That's all. The node is running you can see log outputs in your console. It will download a bit of missing data since the backup was done but it shouldn't take much time.
+
+
+## mainnet
+
+### 1. Clone nearcore project from GitHub
+
+You need to clone [`nearcore`](https://github.com/near/nearcore)
+
+```bash
+$ git clone https://github.com/near/nearcore
+$ git fetch origin --tags
+$ cd nearcore
+```
+
+Checkout to the branch you need if not `master` (default). Latest release is recommended. Please check the [releases page on GitHub](https://github.com/near/nearcore/releases). Current latest is `1.19.0`
+
+```bash
+$ git checkout tags/1.19.0 -b mynode
+```
+
+### 2. Compile nearcore binary
+
+In the `nearcore` folder run the commands:
+
+```bash
+$ cargo build --release --package neard --bin neard
+```
+
+This will start the compilation process, it will take some time depending on your machine power (e.g. i9 8-core CPU, 32 GB RAM, SSD it takes approximately 25 minutes)
```

Comment:
```suggestion
This will start the compilation process and will take some time depending on your machine's CPU power. _(e.g. i9 8-core CPU, 32 GB RAM, SSD takes approximately 25 minutes)_
```